### ISSUE TITLE
Expected Behavior:  Automatic Real-Time Serial Port Change Detection

### ISSUE THREAD

**Issue by ghost** [PeterWone/vscode-arduino-cli#6]
Ok, new thread instead of continuing to hijack the existing threads...

In VSC, I am attaching different controllers and the serial port isn't automatically updating.  What I expect to see is a real-time list of serial ports when I attach and remove controllers.

When I attach a controller, I expect the serial port drop-down to include the new controller's serial port.  But, I don't see it.  I restart VSC and the serial port drop-down has been updated.

I presume this isn't normal behavior and I don't know if there's a setting I need to change to get the expected actions.  I tried this with a WeMos, an Uno, and a NodeMCU, each using a different COM port.  I think 5, 9, and 13...

**Comment by ghost** [comment]
Plus, what should be entered for the "Deployment method"?

Update:  I see the "First time use" description and I don't get a drop-down for this item.

**Comment by ghost** [comment]
Yeah, maybe we can arrange a quick code-share meeting when you're available and get me up and running so I can distinguish between legitimate issues with the extension versus issues with my installation/configuration...perhaps I'm just not quite there yet...

No rush.  I'm currently four hours behind GMT and I'm flexible after work...

**Comment by PeterWone** [comment]
Inadequate documentation and confusing behaviour _are_ legitimate issues, even if they aren't bugs _per se._ I'm just grateful to have fairly sophisticated users who are aware this is alpha code.

Your guess is correct, there is a setting suppressing automatic refresh because it's slooow (~2s). It shouldn't be defaulted to suppressed, I guess. I was contemplating writing my own port detection to see whether I can do it faster in dotnet core.
